### DISCUSSION TITLE
How to mock PIMCore Listings

### DISCUSSION THREAD

**Discussion by alexdrupal** [pimcore/pimcore#10571] [Q&A / Support]
Hello

I would like to test my code that works with DataObject listings. I would like to do it without creating and importing database.

Is it possible to test listing functionality without database in PIM?
If yes - please let me know how.

Thank you!

**Comment by nandorholozsnyak** [comment]
Data Objects are having static methods. We usually wrap these static methods into service classes and we inject these services into other classes and we can mock their results. 
We are using Codeception for these things. We were not able to mock the static methods of DataObjects.
